### PR TITLE
Integrate update start date mutation on cart review page

### PR DESCRIPTION
**Pull request by robinandeer**
## Describe your changes

Remove API route to handle form submission.

Implement hook to handle form submission.

Pass loading state to CartReviewPage.

## Justify why they are needed

Just making the "cart review page" (we should rename it) work with the API.

We should also ask the BE team for a single endpoint to update multiple start dates.

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by robinandeer** [comment]
Current dependencies on/for this PR:
* main
  * **PR #444** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/444" target="_blank">Graphite</a> 
    * **PR #445** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/445" target="_blank">Graphite</a> 
      * **PR #447** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/447" target="_blank">Graphite</a> 
        * **PR #448** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/448" target="_blank">Graphite</a> 
          * **PR #449** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/449" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/449?utm_source=stack-comment).

File: apps/store/src/components/CartReviewPage/useHandleSubmitStartDates.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+    await Promise.all(inputs.map((variables) => updateStartDate({ variables })))
+    onSuccess()
+  }
+
+  return [handleSubmit, result] as const
+}
+
+const isDateInputElement = (value: unknown): value is HTMLInputElement => {
+  return typeof value === 'object' && value !== null && 'valueAsDate' in value
```
Comment (alebedev) [inline apps/store/src/components/CartReviewPage/useHandleSubmitStartDates.ts:42]: Can it be simplified to something like `!!value?.valueAsDate` ?

File: apps/store/src/components/CartReviewPage/useHandleSubmitStartDates.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  const [updateStartDate, result] = useStartDateUpdateMutation()
+
+  const handleSubmit: FormEventHandler<HTMLFormElement> = async (event) => {
+    event.preventDefault()
+
+    const inputs = products.map((product) => {
+      const inputElement = event.currentTarget.elements.namedItem(product.lineId)
+
+      if (!isDateInputElement(inputElement)) {
+        console.error('Unable to update start dates')
```
Comment (alebedev) [inline apps/store/src/components/CartReviewPage/useHandleSubmitStartDates.ts:20]: Not sure we want 3 lines of console.error, every one will come with its own stacktrace

**Review by alebedev — APPROVED** [review]

File: apps/store/src/components/CartReviewPage/useHandleSubmitStartDates.ts (hunk 1 above)
Comment (robinandeer) [reply apps/store/src/components/CartReviewPage/useHandleSubmitStartDates.ts:42]: haha do you think that's simplifying it? 😅 I like to be very explicit, it always takes me 10 sec to understand double negatives

In this case TS doesn't allow us to access "valueAsDate" on the object.

File: apps/store/src/components/CartReviewPage/useHandleSubmitStartDates.ts (hunk 2 above)
Comment (robinandeer) [reply apps/store/src/components/CartReviewPage/useHandleSubmitStartDates.ts:20]: Oh good point -- perhaps those should just be into logs?

Comment (alebedev) [reply apps/store/src/components/CartReviewPage/useHandleSubmitStartDates.ts]: Either that, or concat one detailed error message - might be easier to read Datadog in this case

**Comment by robinandeer** [comment]
**Graphite Merge Job**

Current status: ⏳ Queued to merge

This pull request is currently queued to be merged as part of a stack.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/449?utm_source=stack-comment).


Job Reference: bYOGtcBHRMqrhfixkOY3

File: apps/store/src/components/CartReviewPage/useHandleSubmitStartDates.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    })
+
+    await Promise.all(inputs.map((variables) => updateStartDate({ variables })))
+    onSuccess()
+  }
+
+  return [handleSubmit, result] as const
+}
+
+const isDateInputElement = (value: unknown): value is HTMLInputElement => {
```
Comment (guilhermespopolin) [inline apps/store/src/components/CartReviewPage/useHandleSubmitStartDates.ts:41]: Nice usage of type predicates

**Review by guilhermespopolin — APPROVED** [review]

**Pull request merged**